go/def.bzl

```diff
@@ -126,6 +127,7 @@ TOOLS_NOGO = [
 # check this to determine compatibility.
 RULES_GO_VERSION = "0.35.0"
 
+declare_go_toolchains = _declare_go_toolchains
 declare_toolchains = _declare_toolchains
```

Comment:
`declare_toolchains` isn't documented anywhere, but it is in a "public" file. Could you make it a macro that just calls to two macros under `go/private`, one that defines `toolchain`s and one that defines `go_toolchains`?

Reply:
Mostly yes, but could you also move `declare_X_toolchains` out of `go/defs.bzl`? I would rather choose not to export them publicly, the are only intended to be used by our own generated SDK build files for now. We can always make them public API later if users need them.